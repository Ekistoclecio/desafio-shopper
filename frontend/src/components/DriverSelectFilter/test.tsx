import { screen, waitFor } from '@testing-library/react';

import { DriverSelectFilter } from '.';

import { renderWithTheme } from '@/utils/tests/helpers';
import userEvent from '@testing-library/user-event';

describe('<DriverSelectFilter />', () => {
  it('should render the DriverSelectFilter', () => {
    const { container } = renderWithTheme(
      <DriverSelectFilter
        label="Motorista"
        options={[
          { value: 1, label: 'Motorista 1' },
          { value: 2, label: 'Motorista 2' },
          { value: 3, label: 'Motorista 3' },
          { value: -1, label: 'Todos' },
        ]}
        selectedOption={{ value: 1, label: 'Motorista 1' }}
        onChange={() => {}}
      />,
    );

    expect(screen.getByLabelText('Motorista')).toBeInTheDocument();
    expect(screen.getByRole('combobox')).toBeInTheDocument();
    expect(screen.getByText('Motorista 1')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the DriverSelectFilter with all options', () => {
    renderWithTheme(
      <DriverSelectFilter
        label="Motorista"
        options={[
          { value: 1, label: 'Motorista 1' },
          { value: 2, label: 'Motorista 2' },
          { value: 3, label: 'Motorista 3' },
          { value: -1, label: 'Todos' },
        ]}
        selectedOption={{ value: -1, label: 'Todos' }}
        onChange={() => {}}
      />,
    );

    expect(screen.getByRole('combobox')).toBeInTheDocument();
    expect(screen.getByText('Todos')).toBeInTheDocument();
  });

  it('should call onChange function when a new option is selected', async () => {
    const onChange = jest.fn();

    renderWithTheme(
      <DriverSelectFilter
        label="Motorista"
        options={[
          { value: 1, label: 'Motorista 1' },
          { value: 2, label: 'Motorista 2' },
          { value: 3, label: 'Motorista 3' },
          { value: -1, label: 'Todos' },
        ]}
        selectedOption={{ value: 1, label: 'Motorista 1' }}
        onChange={onChange}
      />,
    );

    expect(screen.getByLabelText('Motorista')).toBeInTheDocument();
    expect(screen.getByRole('combobox')).toBeInTheDocument();
    expect(screen.getByText('Motorista 1')).toBeInTheDocument();

    const combobox = screen.getByRole('combobox');
    await userEvent.click(combobox);
    const newOption = screen.getByText('Motorista 2');
    await userEvent.click(newOption);

    waitFor(() => {
      expect(onChange).toHaveBeenCalledTimes(1);
      expect(onChange).toHaveBeenCalledWith({ value: 2, label: 'Motorista 2' });
    });
  });
});
