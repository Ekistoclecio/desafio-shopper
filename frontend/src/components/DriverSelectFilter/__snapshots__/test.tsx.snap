// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DriverSelectFilter /> should render the DriverSelectFilter 1`] = `
<div
  class="MuiBox-root css-fraksn"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink css-zge7vd-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink="true"
    id="driver-select-filter"
  >
    Motorista
  </label>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-1u4cwut-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  >
    <div
      aria-controls=":r0:"
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="driver-select-filter"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      role="combobox"
      tabindex="0"
    >
      Motorista 1
    </div>
    <input
      aria-hidden="true"
      aria-invalid="false"
      class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
      tabindex="-1"
      value="1"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-w4cd9x"
      >
        <span
          class="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
